subsys/random/rand32_xoroshiro128.c

```diff
@@ -58,8 +58,18 @@ static int xoroshiro128_initialize(struct device *dev)
 		return -EINVAL;
 	}
 
-	if (entropy_get_entropy(dev, (uint8_t *)&state, sizeof(state)) < 0) {
-		return -EINVAL;
+	if (entropy_get_entropy_isr(dev,
```

Comment:
I would do it in the same way the kernel does: https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/init.c#L406, by checking specifically for `ENOTSUP`.

Reply:
sure. I fixed that.